File: src/DynamicLinkLibraries/3DConverters/Abstract3DConverters/IMaterialCreator.cs

using System.Reflection;

namespace Abstract3DConverters
{
    public interface IMaterialCreator
    {

        Assembly Assembly { get; }
        object Create(Image image);


        object Create(Color color);

        object Create(Material material);

        object Create(MaterialGroup material);

        object Create(DiffuseMaterial material);

        object Create(SpecularMaterial material);

        object Create(EmissiveMaterial material);

        void  Add(object group, object value);


        void Set(object material, object color);

        void SetImage(object material, object image);




        void Set(object material, Color color);

        void SetImage(object material, Image image);



    }
}

File: src/TypeScript/TypeScriptBackup/src/Library/Abstract3DConverters/Interfaces/IMaterialCreator.ts
import { ColorTexture } from "../ColorTexture";
import { ImageTexture } from "../ImageTexture";
import { DiffuseMaterial } from "../Materials/DiffuseMaterial";
import { EmissiveMaterial } from "../Materials/EmissiveMaterial";
import { Material } from "../Materials/Material";
import { MaterialGroup } from "../Materials/MaterialGroup";
import { SpecularMaterial } from "../Materials/SpecularMaterial";


export interface IMaterialCreator {
    /// <summary>
    /// Creates of the  image object
    /// </summary>
    /// <param name="image">The image</param>
    /// <returns>The image object</returns>
    createFromImage(obj: any, image: ImageTexture): any;


    /// <summary>
    /// Creates of the color object
    /// </summary>
    /// <param name="color">The color</param>
    /// <returns>The color object</returns>
    createFromColor(obj: any, color: ColorTexture): any;


    /// <summary>
    /// Creates of the material group object
    /// </summary>
    /// <param name="material">The material group</param>
    /// <returns>The  material group object</returns>
    createFromMaterialGroup(group: MaterialGroup): any

    createFromDiffuseMaterial(material: DiffuseMaterial): any

    createFromSpecularMaterial(material: SpecularMaterial): any

    createFromEmissiveMaterial(material: EmissiveMaterial): any

    createFromMaterial(material: Material): any

    createFromEffect(effecty: Material): any

    addMaterialToGroup(group: any, material: any): void


    setImageToEffect(effect: any, image: any): void

    setOpacityToEffect(effect: any, opacity: number): void

}
